documentation/library_guide/kernel_templates/esimd/radix_sort.rst

```diff
@@ -0,0 +1,322 @@
++=============================+==========================================================+
+| ``bool IsAscending``        | Sort order. Ascending: ``true``; Descending: ``false``.  |
++-----------------------------+----------------------------------------------------------+
+| ``std::uint8_t RadixBits``  | Number of bits to sort per a radix sort algorithm pass.  |
+|                             | Only ``8`` is currently supported.                       |
++-----------------------------+----------------------------------------------------------+
+
+
+Parameters
+----------
+
++------------------------------------------------------+------------------------------------------------------------------+
+| Name                                                 | Description                                                      |
++======================================================+==================================================================+
+|  ``q``                                               | SYCL queue to submit the kernel template to.                     |
++------------------------------------------------------+------------------------------------------------------------------+
+|                                                      | The sequences(s) of elements to apply the algorithm to.          |
+|  - ``rng`` (1)                                       | They can be provided as:                                         |
+|  - ``first``, ``last`` (2)                           |                                                                  |
+|  - ``keys_rng``, ``vals_rng`` (3)                    | - ``sycl::buffer`` (1,3),                                        |
+|  - ``keys_first``, ``keys_last``, ``vals_first`` (4) | - ``oneapi::dpl::experimental::ranges::views::all`` (1,3)        |
+|                                                      | - ``oneapi::dpl::experimental::ranges::views::subrange`` (1,3)   |
+|                                                      | - USM pointer (2,4)                                              |
+|                                                      | - ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` (2,4)          |
++------------------------------------------------------+------------------------------------------------------------------+
+|  ``param``                                           | Kernel configuration structure. ``data_per_workitem``,           |
+|                                                      | can be any value among ``32``, ``64``, ``96``,..., ``k * 32``;   |
+|                                                      | ``workgroup_size`` can be only ``64``.                           |
++------------------------------------------------------+------------------------------------------------------------------+
+
+
+Return Value
+------------
+
+``sycl::event`` object representing a status of the algorithm execution.
+
+
+Memory Usage
```

Comment:
Should we discuss this as "Temporary Memory Requirements" instead of "Memory Usage"?
I'm not sure what is best.

Reply:
If they are requirements, then we should label them that way to reduce confusion. I did not read either section as containing a "requirement", so I don't know if we need to change the wording. If we do change the wording, "Temporary Memory Requirements" and "Local Memory Requirements" seems reasonable.

Reply:
I'm not sure the correct terminology necessarily here:
To try to explain further:

It is required that the algorithm be run on a device which has space to allocate the memory as described by the formula.  If they try to run it on a device which doesn't have that data, the algorithm will fail to allocate this data and undefined behavior will occur.

Reply:
That sounds like a requirement to me, but I feel like that is covered by the sentence below the example in Local Memory: "The device must have enough local memory to execute the selected configuration." Maybe we just need to add a similar sentence to the Global Memory section. If "requirements" isn't working we can always go with "parameters".